File: API/Client/OrderSubsidyDTO.js
goog.provide('API.Client.OrderSubsidyDTO');

/**
 * Общее вознаграждение партнеру за DBS-доставку и все скидки на товар:  * по промокодам, купонам и акциям; * по баллам Плюса; * по доставке (DBS).  Передается в валюте заказа. 
 * @record
 */
API.Client.OrderSubsidyDTO = function() {}

/**
 * @type {!API.Client.OrderSubsidyType}
 * @export
 */
API.Client.OrderSubsidyDTO.prototype.type;

/**
 * Сумма субсидии.
 * @type {!number}
 * @export
 */
API.Client.OrderSubsidyDTO.prototype.amount;

File: src/main/scala/org/openapitools/client/api/OrderSubsidyDTO.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import OrderSubsidyDTO._

case class OrderSubsidyDTO (
  `type`: Option[OrderSubsidyType],
/* Сумма субсидии. */
  amount: Option[BigDecimal])

object OrderSubsidyDTO {
  import DateTimeCodecs._

  implicit val OrderSubsidyDTOCodecJson: CodecJson[OrderSubsidyDTO] = CodecJson.derive[OrderSubsidyDTO]
  implicit val OrderSubsidyDTODecoder: EntityDecoder[OrderSubsidyDTO] = jsonOf[OrderSubsidyDTO]
  implicit val OrderSubsidyDTOEncoder: EntityEncoder[OrderSubsidyDTO] = jsonEncoderOf[OrderSubsidyDTO]
}
